src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/CSharpExpressionCompiler.cs

```diff
@@ -89,33 +147,49 @@ internal override DkmCompilerId CompilerId
                     localSignatureToken);
             }
 
-            var previous = appDomain.GetMetadataContext<CSharpMetadataContext>();
+            var previous = getMetadataContext(appDomain);
+            if (!previous.Matches(metadataBlocks))
+            {
+                previous = null;
+            }
+            if (previous != null && previous.ModuleVersionId != moduleVersionId)
```

Comment:
If the fix is disabled (if `kind == AllAssemblies`), we should not be setting or checking `previous.ModuleVersionId`.